### PR TITLE
Set minimum version for NCEPLIB-bufr

### PR DESCRIPTION
**Pull request by rmclaren**
## Description

Sets the minimum required version for NCEPLIB_bufr to 12.0.0.

## Issue(s) addressed

#1380

### CODE REVIEW

**Comment by CoryMartin-NOAA** [comment]
@rmclaren will this mean that your BUFR code now won't build on NOAA HPC without spack-stack 1.5.0 even with this change?

**Comment by rmclaren** [comment]
@CoryMartin-NOAA Should build, but won't include BUFR code if relevant version of NCEPLIB-bufr is not found...

**Comment by rmclaren** [comment]
Can anyone see why builds are failing... I don't have permissions to see whats going on...

**Comment by rmclaren** [comment]
Nevermind. Figured it out...

**Review by danholdaway — APPROVED** [review]

**Comment by danholdaway** [comment]
> Nevermind. Figured it out...

Sorry I led you astray with the MINIMUM_VERSION thing.

I'm fine with merging this instead of merging the revert PR.

**Review by srherbener — COMMENTED** [review]
Has this been tested with an environment that does not have version 12.0.0? Make sure all the code sections and tests are shut off when the proper versions of bufr is not availeable.

Thanks!

**Comment by fcvdb** [comment]
I tried to build the IODA bundle on Orion with `spack-stack 1.4.0` (bufr/11.7.1) and the `ecbuild` configuration failed with this error message:

```
CMake Error at /work/noaa/epic-ps/role-epic/spack-stack/spack-stack-1.4.0/envs/unified-env-v2/install/gcc/10.2.0/ecbuild-3.7.2-ptjxnc5/share/ecbuild/cmake/ecbuild_add_executable.cmake:248 (target_link_libraries):
  Target "print_queries.x" links to:
    bufr::bufr_4
  but the target was not found. 
```

**Comment by rmclaren** [comment]
Shoot, I'll take a look.

**Comment by rmclaren** [comment]
@fcvdb Latest checkin should fix this problem.

**Comment by PatNichols** [comment]
> Has this been tested with an environment that does not have version 12.0.0? Make sure all the code sections and tests are shut off when the proper versions of bufr is not availeable.
> 
> Thanks!

@srherbener I am testing on aws. There are some bugs in our cmake that are now apparent. There is one test that runs that shouldn't and some others I am looking at. There are 2 ways to tests. Use spack-stack-1.5 but unload the bufr module or use spack-stack-1.4.0. I uses the former though the latter might be closer to what users are doing.

File: test/CMakeLists.txt [hunk 1]
```diff
@@ -1901,6 +1901,7 @@ if( iodaconv_eccodes_ENABLED )
                            --date 2021-02-01T12:00:00Z"
                            wmo_raob_double.nc4 ${IODA_CONV_COMP_TOL_ZERO})
 
+if( iodaconv_bufr_ENABLED )
```
[the remark this answers is not in the document]
Comment (fcvdb) [reply test/CMakeLists.txt:1904]: This is for consistency with the condition `if( iodaconv_bufr_ENABLED )` at line [197](https://github.com/JCSDA-internal/ioda-converters/blob/3e8d7c03d4fc6f5caeb8e561ff03da3828949b12/test/CMakeLists.txt#L197C5-L197C26) that prevents installing the input file [testinput/amdar_wmo_multi.bufr](https://github.com/JCSDA-internal/ioda-converters/blob/3e8d7c03d4fc6f5caeb8e561ff03da3828949b12/test/CMakeLists.txt#L272)

As a result the tests `test_iodaconv_amdar, test_iodaconv_buoy,  test_iodaconv_ship, and test_iodaconv_synop ` fail for lack of input file.

File: test/CMakeLists.txt [hunk 2]
```diff
@@ -1955,6 +1956,8 @@ if( iodaconv_eccodes_ENABLED )
 
 endif()
 
+endif()
```
[the remark this answers is not in the document]
Comment (fcvdb) [reply test/CMakeLists.txt:1959]: One `endif()` for `if( iodaconv_eccodes_ENABLED )` (line [1885](https://github.com/JCSDA-internal/ioda-converters/blob/3e8d7c03d4fc6f5caeb8e561ff03da3828949b12/test/CMakeLists.txt#L1885C1-L1885C31))

One `endif` for `if( iodaconv_bufr_ENABLED )` (line [1904](https://github.com/JCSDA-internal/ioda-converters/blob/3e8d7c03d4fc6f5caeb8e561ff03da3828949b12/test/CMakeLists.txt#L1904C1-L1904C28)).

**Review by emilyhcliu — APPROVED** [review]

File: test/CMakeLists.txt (hunk 2 above)
[the remark this answers is not in the document]
Comment (rmclaren) [reply test/CMakeLists.txt:1959]: These should not be nested...

File: test/CMakeLists.txt (hunk 1 above)
[the remark this answers is not in the document]
Comment (rmclaren) [reply test/CMakeLists.txt:1904]: If these other tests are dependent on those BUFR files, a better solution would be to, for example, install the amdar_wmo_multi.bufr file regardless of iodaconv_bufr_ENABLED flag. Or check the eccodes and "bufr" flag with an "or" statement and install if either one or the other are installed.

**Review by PatNichols — APPROVED** [review]
@srherbener @rmclaren I am getting all but one tests failing on the MacOS. It appears that bufr is building properly under spack-stack. I don't think this a problem with this PR. I am approving this and we can discover what the problem is with spack-stack later.

**Comment by PatNichols** [comment]
@fcvdb did this pass on Orion?

**Pull request merged**